fossunited/www/book-conference-ticket/index.py

```diff
@@ -0,0 +1,185 @@
+            ),
+            "student_ticket_price": values.get(
+                "student_ticket_price"
+            ),
+            "total_amount": values.get("total_amount"),
+        }
+    )
+
+    ticket.insert()
+    ticket.order_id = makeOrderId(ticket.name, ticket.total_amount)
+    ticket.save()
+
+    return {
+        "ticket_id": ticket.name,
+        "order_id": ticket.order_id,
+        "amount": ticket.total_amount,
+        "razorpay_key": frappe.db.get_single_value(
+            "Razorpay Keys", "rzp_key"
+        ),
+        "currency": "INR",
+        "fullname": ticket.first_name + " " + ticket.last_name,
+        "email": ticket.email,
+    }
+
+
+doc = frappe.db.get_all(
+    "Conference Tickets",
+    fields=[
+        "order_id",
+        "payment_id",
+        "first_name",
+        "last_name",
+        "email",
+        "general_ticket_price",
+        "student_ticket_price",
+        "general_tickets",
+        "student_tickets",
+        "total_amount",
+    ],
+)
+
+
+@frappe.whitelist()
+def makeOrderId(ticket_id, total_amount):
+    rzp_basic_auth = f"Basic {createBase()}"
+
+    order = make_post_request(
+        f"https://api.razorpay.com/v1/orders",
+        headers={
+            "Authorization": rzp_basic_auth,
+            "Content-Type": "application/json",
+        },
+        data=json.dumps(
+            {
+                "amount": f"{total_amount * 100}",
+                "currency": "INR",
+                "notes": {"ticket_id": ticket_id},
+            }
+        ),
+    )
+    return order["id"]
+
+
+# Function to auomate the formatting of api_key:api_secret
+def createBase():
+    key_id = frappe.db.get_single_value("Razorpay Keys", "rzp_key")
+    secretKey = frappe.get_single("Razorpay Keys")
+    key_secret = secretKey.get_password("key_secret")
+    basic_auth = base64.b64encode(
+        f"{key_id}:{key_secret}".encode()
+    ).decode()
+
+    return basic_auth
+
+
+# Function to update captured payment state
+@frappe.whitelist()
+def capture_payment():
+    doc = frappe.get_doc(
+        "Conference Tickets", frappe.form_dict.ticket_id
+    )
+
+    if not doc.payment_captured:
+        doc.razorpay_payment_id = frappe.form_dict.razorpay_payment_id
+        doc.razorpay_signature = frappe.form_dict.razorpay_signature
+
+        razorpay_key = frappe.db.get_single_value(
+            "Razorpay Keys", "rzp_key"
+        )
+
+        order = make_get_request(
+            f"https://api.razorpay.com/v1/orders/{doc.order_id}",
+            headers={
+                "Authorization": basic_auth,
+                "Content-Type": "application/json",
+            },
+        )
+
+        doc.payment_captured = (
+            True if order["status"] == "paid" else False
+        )
+
+        if doc.payment_captured:
+            if doc.total_amount != order["amount_paid"] / 100:
+                frappe.throw(
+                    f"invalid amounts {doc.total_amount} {order['amount_paid'] / 100}"
```

Comment:
write a better error message. 'Total Amount Mismatch' or something. And let the user know if the order was captured or not